core/trino-main/src/test/java/io/trino/sql/ir/optimizer/TestSimplifyContinuousInValues.java

```diff
@@ -0,0 +1,178 @@
+                .describedAs("non-null continuous values")
+                .isEqualTo(Optional.of(
+                        new Between(new Reference(BIGINT, "x"), new Constant(BIGINT, 1L), new Constant(BIGINT, 3L))));
+
+        assertThat(optimize(
+                new In(
+                        new Reference(BIGINT, "x"),
+                        ImmutableList.of(
+                                new Constant(BIGINT, 1L),
+                                new Constant(BIGINT, 2L),
+                                new Constant(BIGINT, 2L),
+                                new Constant(BIGINT, 3L)))))
+                .describedAs("repeated continuous values")
+                .isEqualTo(Optional.empty());
+
+        assertThat(optimize(
+                new In(
+                        new Reference(BIGINT, "x"),
+                        ImmutableList.of(
+                                new Constant(BIGINT, 1L),
+                                new Constant(BIGINT, 2L),
+                                new Constant(BIGINT, 4L),
+                                new Constant(BIGINT, 5L)))))
+                .describedAs("discontinuous values")
+                .isEqualTo(Optional.empty());
+
+        assertThat(optimize(
+                new In(
+                        new Reference(BIGINT, "x"),
+                        ImmutableList.of(new Constant(BIGINT, Long.MAX_VALUE), new Constant(BIGINT, Long.MIN_VALUE)))))
+                .describedAs("overflow handling")
+                .isEqualTo(Optional.empty());
+
+        assertThat(optimize(
+                new In(new Reference(DOUBLE, "x"), ImmutableList.of(new Constant(DOUBLE, 1.0), new Constant(DOUBLE, 2.0), new Constant(DOUBLE, 3.0)))))
+                .describedAs("unsupported type")
+                .isEqualTo(Optional.empty());
+
+        assertThat(optimize(
+                new In(
+                        new Reference(REAL, "x"),
+                        ImmutableList.of(
+                                new Constant(REAL, Reals.toReal(1.0f)),
+                                new Constant(REAL, Reals.toReal(2.0f)),
+                                new Constant(REAL, Reals.toReal(3.0f))))))
+                .describedAs("unsupported type")
+                .isEqualTo(Optional.empty());
+    }
+
+    @Test
+    public void verifySupportedTypeAssumptions()
+    {
+        for (Type type : supportedTypes()) {
+            assertThat(type.getJavaType()).isEqualTo(long.class);
+
+            BlockBuilder blockBuilder = type.createBlockBuilder(null, 201);
+            for (long value = -100; value <= 100; value++) {
+                type.writeLong(blockBuilder, value);
+            }
+            Block block = blockBuilder.build();
+
+            for (int i = 0; i < block.getPositionCount(); i++) {
+                assertThat(type.getLong(block, i)).isEqualTo(i - 100);
+            }
+        }
+    }
```

Comment:
I guess I was referring to adding a test of the rule for each type. It's not enough to ensure that the types support long, but that the representation satisfies the assumptions that logical values are contiguous in the representation space.

Reply:
I've updated the test to do that now.
I've also added a couple of commits to implement getRange/getPreviousValue/getNextValue on TimeType and ShortDecimalType for usage in testing here.